js/directives/lucca/user-picker.js

```diff
@@ -0,0 +1,322 @@
+	"class=\"lui regular nguibs-ui-select\" on-select=\"addSelectedUser()\" on-remove=\"onRemove()\" ng-disabled=\"controlDisabled\">" +
+	"<ui-select-match placeholder=\"VAR_TRAD Sélectionner un utilisateur...\">{{$item.firstName}} {{$item.lastName}} " +
+	"<span ng-if=\"$item.hasHomonyms\" ng-repeat=\"property in properties\">&lt{{getProperty($item, property)}}&gt</span>" +
+	"<span ng-if=\"$item.isFormerEmployee\">&lt;VAR_TRAD Parti(e) le {{$item.dtContractEnd | luifMoment: 'll'}}&gt;</span>" +
+	"</ui-select-match>" +
+	uiSelectChoicesTemplate +
+	"</ui-select>";
+
+
+	angular.module('lui.directives')
+	.directive('luidUserPicker', function () {
+		return {
+			restrict: 'E',
+			controller: "luidUserPickerController",
+			template: userPickerTemplate,
+			require: "luidUserPicker",
+			scope: {
+				showFormerEmployees: "=",
+				onSelect: "&",
+				onRemove: "&",
+				controlDisabled: "=",
+				useCustomFilter: "=",
+				customFilter: "&",
+				appId: "@",
+				operation: "@",
+				homonymsProperties: "@"
+			},
+			link: function (scope, elt, attrs, ctrls) {
+				if (attrs.homonymsProperties) {
+					scope.properties = attrs.homonymsProperties.split(',');
+				}
+				else {
+					scope.properties = DEFAULT_HOMONYMS_PROPERTIES;
+				}
+			}
+		};
+	})
+
+	.directive('luidUserPickerMultiple', function () {
+		return {
+			restrict: 'E',
+			controller: "luidUserPickerController",
+			template: userPickerMultipleTemplate,
+			require: "luidUserPickerController",
+			scope: {
+				showFormerEmployees: "=",
+				onSelect: "&",
+				onRemove: "&",
+				controlDisabled: "=",
+				useCustomFilter: "=",
+				customFilter: "&",
+				appId: "@",
+				operation: "@",
+				homonymsProperties: "@"
+			},
+			link: function (scope, elt, attrs, ctrls) {
+				if (attrs.homonymsProperties) {
+					scope.properties = attrs.homonymsProperties.split(',');
+				}
+				else {
+					scope.properties = DEFAULT_HOMONYMS_PROPERTIES;
+				}
+			}
+		};
+	})
+
+	.controller("luidUserPickerController", ['$scope', '$http', 'moment', '$timeout', '$q', function ($scope, $http, moment, $timeout, $q) {
+		var timeout = {};
+		var promise;
+		var maxUsers = 0;
+		var selectedUsersCount = 0;
+
+		$scope.user = {};
+		$scope.selected = {};
+		$scope.selected.users = [];
+
+		var getUsers = function(init, input) {
+			var formerEmployees = "formerEmployees=" + ($scope.showFormerEmployees ? "true" : "false");
+			var limit = "&limit=";
+			var clue = "&clue=" + input;
+			var query = "/api/v3/users/find?" + (init ? "" : (clue + "&")) + formerEmployees;
+
+			if (promise) {
+				promise.then(function (response) {}); // do nothing
+			}
+
+			// Cancel previous timeout
+			if (timeout.count) {
+				$timeout.cancel(timeout.count);
+			}
+
+			// If we use a UserPickerMultiple or a custom filter, we ask for all the users
+			if ((selectedUsersCount !== 0) || ($scope.useCustomFilter)) {
+				limit += MAGIC_NUMBER_maxUsers;
+			}
```

Comment:
in case of a multi-select, you don't need *ALL* the users, you need `MAX_COUNT` + 1 /* for pagination */ + selectedCnt`

also in case of a simple select when there is already someone selected you need to ask for `MAX_COUNT + 1 + 1` in case the list of return users contains the selected user